frontend/packages/ceph-storage-plugin/src/components/modals/storage-pool-modal/storage-pool-modal.tsx

```diff
@@ -0,0 +1,391 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+
+import {
+  Switch,
+  Dropdown,
+  DropdownToggle,
+  DropdownItem,
+  EmptyState,
+  EmptyStateIcon,
+  EmptyStateBody,
+  ActionGroup,
+  Button,
+} from '@patternfly/react-core';
+import {
+  CaretDownIcon,
+  LockIcon,
+  CheckCircleIcon,
+  ExclamationCircleIcon,
+  UnknownIcon,
+} from '@patternfly/react-icons';
+
+import {
+  createModalLauncher,
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+  ModalComponentProps,
+  ModalFooter,
+} from '@console/internal/components/factory/modal';
+import { apiVersionForModel } from '@console/kubevirt-plugin/integration-tests/tests/utils/selectors';
+import {
+  HandlePromiseProps,
+  withHandlePromise,
+} from '@console/internal/components/utils/promise-component';
+import { k8sCreate, k8sGet } from '@console/internal/module/k8s/resource';
+
+import { CephClusterKind, StoragePoolKind, OCSStorageClusterKind } from '../../../types';
+import { CephBlockPoolModel, OCSServiceModel } from '../../../models';
+import { CEPH_STORAGE_NAMESPACE, OCS_INTERNAL_CR_NAME } from '../../../constants/index';
+
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { replicas, modalTitle, modalDescription, notReadyDesc } from './storage-pool-constant';
+
+import './_storage-pool-modal.scss';
+
+const LoadingComponent: React.FC = () => (
+  <span className="pf-c-spinner" role="progressbar" aria-valuetext="Loading...">
+    <span className="pf-c-spinner__clipper" />
+    <span className="pf-c-spinner__lead-ball" />
+    <span className="pf-c-spinner__tail-ball" />
+  </span>
+);
+
+const PoolStatusComponent: React.FC<PoolStatusComponentProps> = ({ status, name, error = '' }) => {
+  return (
+    <div>
+      {status === 'progress' && (
+        <EmptyState>
+          <EmptyStateIcon icon={LoadingComponent} />
+          <EmptyStateBody>Pool {name} creation in progress</EmptyStateBody>
+        </EmptyState>
+      )}
+      {status === 'created' && (
+        <EmptyState>
+          <EmptyStateIcon icon={CheckCircleIcon} className="ceph-storage-pool__check-icon" />
+          <EmptyStateBody>Pool {name} was successfully created</EmptyStateBody>
+        </EmptyState>
+      )}
+      {status === 'failed' && (
+        <EmptyState>
+          <EmptyStateIcon icon={ExclamationCircleIcon} className="ceph-storage-pool__error-icon" />
+          <EmptyStateBody>
+            {error || `An error occurred, Pool ${name} was not created`}
+          </EmptyStateBody>
+        </EmptyState>
+      )}
+      {status === 'timeout' && (
+        <EmptyState>
```

Comment:
```
        <EmptyState>
          <EmptyStateIcon icon={ExclamationCircleIcon} className="ceph-storage-pool__error-icon" />
          <EmptyStateBody>
            {error || `An error occurred, Pool ${name} was not created`}
          </EmptyStateBody>
        </EmptyState>
```
It make sense to create a common component for this, since its used at many places. for eg. 
`<EmptyStateContainer icon={} message={} /> WDYT?

Reply:
ack